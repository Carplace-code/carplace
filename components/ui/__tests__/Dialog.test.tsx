// components/__tests__/Dialog.test.tsx
import { render } from "@testing-library/react";

import {
  Dialog,
  DialogClose,
  DialogContent,
  DialogDescription,
  DialogFooter,
  DialogHeader,
  DialogTitle,
  DialogTrigger,
} from "../dialog";

describe("Dialog", () => {
  it("matches snapshot", () => {
    const { container } = render(
      <Dialog>
        <DialogTrigger>
          <button type="button">Open Dialog</button>
        </DialogTrigger>
        <DialogContent>
          <DialogHeader>
            <DialogTitle>Example Title</DialogTitle>
            <DialogDescription>Example description.</DialogDescription>
          </DialogHeader>
          <div>Dialog body content</div>
          <DialogFooter>
            <DialogClose>
              <button type="button">Close</button>
            </DialogClose>
          </DialogFooter>
        </DialogContent>
      </Dialog>,
    );

    expect(container).toMatchSnapshot();
  });
});
