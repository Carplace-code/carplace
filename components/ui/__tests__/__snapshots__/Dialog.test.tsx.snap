// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Dialog > matches snapshot 1`] = `
<div>
  <button
    aria-controls="radix-«r0»"
    aria-expanded="false"
    aria-haspopup="dialog"
    data-slot="dialog-trigger"
    data-state="closed"
    type="button"
  >
    <button
      type="button"
    >
      Open Dialog
    </button>
  </button>
</div>
`;
